casepro/msgs/forms.py

```diff
@@ -79,3 +80,45 @@ def clean_keywords(self):
     class Meta:
         model = Label
         fields = ('name', 'description', 'is_synced', 'keywords', 'groups', 'field_test', 'ignore_single_words')
+
+
+class FaqForm(forms.ModelForm):
+
+    question = forms.CharField(label=_("Question"), max_length=140)
+    answer = forms.CharField(label=_("Answer"), max_length=140)
+    language = forms.CharField(label=_("Language"), max_length=3)
+    # limit the parent choices to FAQs that have a ForeignKey parent that is None
+    parent = forms.ModelChoiceField(queryset=FAQ.objects.filter(parent=None), required=False)
+    labels = forms.ModelMultipleChoiceField(queryset=Label.objects.filter(), required=False, help_text=_(
+        "If a Parent is selected, the labels will be copied from the Parent FAQ"))
+
+    def clean_language(self):
+        language = self.cleaned_data['language'].strip()
+        if not is_valid639_2(language):
+            raise forms.ValidationError(_("Language must be valid a ISO-639-2 code"))
+        return language
+
+    def clean_labels(self):
+        if 'labels' in self.cleaned_data and len(self.cleaned_data['labels']) != 0:
+            labels = self.cleaned_data['labels']
+        else:
+            labels = None
+
+        if 'parent' in self.cleaned_data:
+            parent = self.cleaned_data['parent']
+        else:
+            parent = None
+
+        if parent is None and labels is None:
+            raise forms.ValidationError(_("Labels are required if no Parent is selected"))
+
+        if parent is not None:
+            labels = parent.labels.all()
```

Comment:
Would seem better to derive labels from parent rather than trying to keep parent and children in sync